design/feature-flags.md

```diff
@@ -0,0 +1,47 @@
+# Feature Flags
+
+Status: Draft
+
+Some features may take a while to get fully implemented, and we don't necessarily want to have long-lived feature branches
+A simple feature flag implementation allows code to be merged into master, but not used unless a flag is set.
+
+## Goals
+
+- Allow unfinished features to be present in Velero releases, but only enabled when the associated flag is set.
+
+## Non Goals
+
+- A robust feature flag library.
+
+## Background
+
+When considering the [CSI integration work](https://github.com/heptio/velero/pull/1661), the timelines involved presented a problem in balancing a release and longer-running feature work.
+A simple implementation of feature flags can help protect unfinished code while allowing the rest of the changes to ship.
+
+## High-Level Design
+
+A new command line flag, `--features` will be added to the root `velero` command.
+
+`--features` will accept a comma-separated list of features, such as `--features EnableCSI,Replication`.
+Each feature listed will correspond to a key in a map in `pkg/features/features.go` defining whether a feature should be enabled.
+
+Any code implementing the feature would then import the map and look up the key's value.
+
+For the Velero client, a `features` key can be added to the config.json file for more convenient client invocatons.
+
+## Detailed Design
```

Comment:
We might want to drop all the placeholder stuff after this, so it's less confusing to read this proposal